### ISSUE TITLE
Uglify error

### ISSUE THREAD

**Issue by ShunKeJiang** [aws/aws-iot-device-sdk-js#295]
I just tried NPM, but there is something wrong with uglify directly. Maybe I can change it to uglify-es3. If you have time, you can change your document. thank you

**Comment by KaibaLopez** [comment]
Hi @ShunKeJiang 
Not sure what you mean here,can you be a bit more specific?
what are you trying to do and what error are you getting? How can I reproduce the error ?

**Comment by ShunKeJiang** [comment]
@KaibaLopez Uglify doesn't seem to work when aws- tot-sdk-browser-bundle compression is minimal. I tried many times. Finally I moved on to uglifyjs but he would recognize the ES6 error. Finally, I successfully made an installation package with ES5. You can try it

**Comment by KaibaLopez** [comment]
Hi @ShunKeJiang, correct me if I'm wrong but Uglify does not support ES6, so using ES5 is expected: 
[Uglify Documentation](https://www.npmjs.com/package/uglify-js)
I still feel like I'm missing on something but it sounds like you've found a solution to your own problem? Or what exactly are you suggesting we change in the document?

**Comment by ShunKeJiang** [comment]
@KaibaLopez If you use your official document compression method, it is not feasible. The compressed file is empty. I tried to find it and finally found that uglify.js is feasible

**Comment by ShunKeJiang** [comment]
@KaibaLopez 
![s0](https://user-images.githubusercontent.com/40446492/69111447-90ee0680-0ab8-11ea-80ae-6a0111d46dce.jpg)

**Comment by ShunKeJiang** [comment]
@KaibaLopez 
![s0](https://user-images.githubusercontent.com/40446492/69111480-a95e2100-0ab8-11ea-8906-5e816e5832f8.jpg)

**Comment by KaibaLopez** [comment]
ahh I see...
Yes, so the idea is that by the time you've reached that point, you would've gone through the browserify process, so you'd have to have ran the command:
`npm run-script browserize`
That would create the aws-iot-sdk-browser-bundle.js file and then you can use the uglify command on it.
Let me know if I'm missing something again though, or if this also doesn't work for you. For me it compressed the original browser-bundle.js from a 5.2mb file to a 2.3MB min.js

**Comment by ShunKeJiang** [comment]
@KaibaLopez You can try reconstructing one from the document.

**Comment by KaibaLopez** [comment]
Hi @ShunKeJiang sorry if I'm misunderstanding or not being clear, but that's what I did and for me, when I followed the instructions on the read me, it worked just fine. It is unclear to me from your answers what steps or settings you had before trying uglify, so I can't know what went wrong for you.

That being said, the fact that you had trouble with it may mean that the documentation could be better written....  Which part would you say we should change, as in, which step would you say we are missing or which instruction was unclear to you? Did you try what I said on the last comment? did that also not work?

**Comment by ShunKeJiang** [comment]
@KaibaLopez I hope you can try it again with Windows. Maybe it is my computer, but I have tested two computers. Cheers!

**Comment by KaibaLopez** [comment]
alright! sorry for all the confusion, 
You are right, there seems to be a bug with uglify on windows. This seem to be an issue on uglify though and not specifically to our sdk... I'll look for a workaround for now

**Comment by ShunKeJiang** [comment]
@KaibaLopez I now is only temporary compression, but I test uglify js will require will NPM err - 1, I finally through uglify - es. Js solves the compression, but I found that there is the require in, perhaps you require is encapsulated a function, but on the ground floor is still a ES6 function body, I believe you can get rid of the BUG, and alternative, then I will replace my packaging.

**Comment by tronious** [comment]
I haven't read this thread in its entirety.  BUT I had the same exact issue.  Something to do with the dependences...and const gets ignored.

The way I worked around it was to take the .js files from the node_module, run them through browserify and create a minified version of both files combine.

Then instead of using the node module...I just imported the minified sdk file.

This may/may not help you...but I can tell you I spent a good deal of time on deal with Uglify failing.  My root cause may/may not be the same as yours.  Perhaps this will help...perhaps not
